### PR TITLE
feat: Implement lxa review command for reviewer-centric PR queue

### LINKED ISSUE #76: Add lxa review command for reviewer-centric PR queue
## Summary

Add a new `lxa review` command that provides a reviewer-centric view of your GitHub review queue. While `lxa pr list` answers "What's happening with my PRs?", `lxa review` answers "What PRs need my review attention?"

## Motivation

As a code reviewer, I want to quickly see:
1. PRs where I'm requested but haven't reviewed yet
2. PRs where I reviewed but there's new activity I need to look at

The existing `lxa pr list --reviewer me` shows PRs where I'm requested, but doesn't help me understand my review status or prioritize by wait time.

## Proposed Solution

A new `lxa review` command with:

- **Default view**: Only PRs needing my action (pending review or re-review needed)
- **History string**: Reuse the signature feature from `lxa pr list`, showing the timeline from my reviewer perspective
- **Status column**: `review`, `re-review`, `hold`, `approved`
- **Wait time**: How long the PR has been waiting specifically for me
- **Priority sorting**: Longest-waiting reviews first

### Example Output

```
 Repo           PR    History   Status      Wait    CI       💬   Author   Last
─────────────────────────────────────────────────────────────────────────────────
 owner/repo    #142   OHrF      re-review   3h      green     2   alice    1h ago
 owner/repo2   #51    OH        review      2d      green    --   bob      2h ago
```

### CLI Options

```
lxa review              # Show PRs needing my review action (default)
lxa review --all        # Include hold/approved PRs
lxa review --author USER
lxa review --repo OWNER/REPO
lxa review --board NAME
lxa review --title
```

## Design Document

See `.pr/review-command-design.md` for full technical design.

## Acceptance Criteria

- [ ] `lxa review` shows pending and re-review PRs
- [ ] `lxa review --all` includes hold and approved PRs
- [ ] History string shows reviewer perspective (lowercase = my actions)
- [ ] Wait time computed correctly for each status
- [ ] Sorted by status priority, then wait time
- [ ] All filter options work (--author, --repo, --board, --limit, --title)

### PR DESCRIPTION
**Pull request by jpshackelford**
## Summary

Adds design document for a new `lxa review` command that provides a reviewer-centric view of your GitHub review queue.

Closes #76

### Core Concept
While `lxa pr list` answers "What's happening with my PRs?", `lxa review` answers "What PRs need my review attention?"

### Key Features
- **Default view**: Only PRs needing your action (pending review or re-review)
- **History string**: Reuses the signature feature, showing timeline from reviewer perspective
- **Status column**: `review`, `re-review`, `hold`, `approved`
- **Wait time**: How long the PR has been waiting specifically for you
- **Priority sorting**: Longest-waiting reviews surface first

### Example Output
```
 Repo           PR    History   Status      Wait    CI       💬   Author   Last
─────────────────────────────────────────────────────────────────────────────────
 owner/repo    #142   OHrF      re-review   3h      green     2   alice    1h ago
 owner/repo2   #51    OH        review      2d      green    --   bob      2h ago
```

## Implementation Plan

The design doc outlines 4 milestones:
1. Core data model and status computation logic
2. GitHub API integration
3. CLI command implementation
4. Polish and documentation

## Files Changed

- `.pr/review-command-design.md` - Full design document

---
*This PR was created by an AI assistant (OpenHands) on behalf of the user.*

### CODE REVIEW

**Comment by jpshackelford** [comment]
@openhands Please implement the design.

File: .pr/review-command-design.md [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+| Status | Meaning | Your Action | Wait = |
+|--------|---------|-------------|--------|
+| `review` | Requested, not yet reviewed | Do initial review | Time since requested |
+| `re-review` | New commits since your last review | Re-review changes | Time since new commits |
+| `hold` | You requested changes, author hasn't pushed | Wait | Time since you requested changes |
+| `approved` | You approved, not yet merged | None | Time since you approved |
+
+### 4. Visual Styling
+
+**Status colors:**
```
[the remark this answers is not in the document]
Comment (jpshackelford) [reply .pr/review-command-design.md:146]: Acknowledged as a UX improvement for future iteration. The current semantics are documented in the design doc and acceptable for v1.

File: .pr/review-command-design.md [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+
+| Code | Meaning |
+|------|---------|
+| `o/O` | Opened |
+| `h/H` | Help requested (review requested) |
+| `r/R` | Review with changes requested |
+| `a/A` | Approved |
+| `c/C` | Comment |
+| `f/F` | Fix (commits pushed after review) |
+| `m/M` | Merged |
```
[the remark this answers is not in the document]
Comment (jpshackelford) [reply .pr/review-command-design.md:122]: Good observation about the convention difference. This can be addressed in user documentation/help text in a future iteration.

File: .pr/review-command-design.md [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+    # I requested changes - check if author pushed since
+    commits_after_review = [e for e in timeline_events
+                           if e.action == ActionType.FIX
+                           and e.timestamp > last_review.timestamp]
+    
+    if commits_after_review:
+        return (ReviewStatus.RE_REVIEW, commits_after_review[0].timestamp)
+    
+    return (ReviewStatus.HOLD, last_review.timestamp)
+```
```
[the remark this answers is not in the document]
Comment (jpshackelford) [reply .pr/review-command-design.md:230]: These edge cases (dismissed reviews, stale reviews, team requests, draft PRs) are valid considerations for v2. The current implementation provides pragmatic v1 functionality.

File: .pr/review-command-design.md [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+
+### Milestone 1: Core Data Model and Status Logic
+
+**Goal:** Implement the reviewer status computation logic.
+
+**Files:**
+- `src/review/models.py` - `ReviewStatus` enum and `ReviewInfo` dataclass
+- `src/review/status.py` - Status computation logic
+- `tests/review/test_status.py` - Unit tests for status computation
+
```
[the remark this answers is not in the document]
Comment (jpshackelford) [reply .pr/review-command-design.md:270]: Good suggestions for test scenarios. The implementation includes tests for core status transitions. Additional edge case tests can be added as the feature matures.

File: .pr/review-command-design.md [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+    
+    return (ReviewStatus.HOLD, last_review.timestamp)
+```
+
+### 7. Data Fetching
+
+#### GitHub Search Queries
+
+```python
+def fetch_review_queue(reviewer: str, repos: list[str] | None) -> list[ReviewInfo]:
```
[the remark this answers is not in the document]
Comment (jpshackelford) [reply .pr/review-command-design.md:237]: Noted for future work. Rate limiting, pagination beyond 100 PRs, and error handling for deleted content are valid v2 enhancements.

File: .pr/review-command-design.md [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+    """Fetch PRs for reviewer's queue."""
+    
+    # Query 1: PRs where reviewer is requested (pending reviews)
+    requested_query = f"is:pr is:open review-requested:{reviewer}"
+    
+    # Query 2: PRs reviewer has reviewed (for re-review detection)
+    reviewed_query = f"is:pr is:open reviewed-by:{reviewer}"
+    
+    # Combine results, deduplicate by PR
+    # Process each PR to compute ReviewInfo
```
[the remark this answers is not in the document]
Comment (jpshackelford) [reply .pr/review-command-design.md:247]: Acknowledged - GitHub search eventual consistency is a known limitation, acceptable for a review queue tool.

File: .pr/review-command-design.md [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+Showing 2 PRs needing your review
+```
+
+### Full View (`--all`)
+
+Adds PRs where the ball is in the author's court:
+
+```
+ Repo           PR    History   Status      Wait    CI       💬   Author   Last
+─────────────────────────────────────────────────────────────────────────────────
```
[the remark this answers is not in the document]
Comment (jpshackelford) [reply .pr/review-command-design.md:50]: Minor wording suggestion noted. Can be refined in a future UX pass.

File: .pr/review-command-design.md [hunk 8]
```diff
@@ -0,0 +1,318 @@
...
+**Goal:** Finalize UX details and add documentation.
+
+**Tasks:**
+- Add help text for all options
+- Update README with `lxa review` usage
+- Add to AGENTS.md knowledge base
+
+**Acceptance Criteria:**
+- `lxa review --help` shows clear usage info
+- Documentation covers common workflows
```
[the remark this answers is not in the document]
Comment (jpshackelford) [reply .pr/review-command-design.md:318]: Agreed - combining design doc and implementation in one PR is pragmatic for this scope.

File: src/review/status.py [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+        if timeline_events:
+            return (ReviewStatus.REVIEW, timeline_events[0].timestamp)
+        # Edge case: no events at all
+        return (ReviewStatus.REVIEW, datetime.now(tz=None))
+
+    last_review = my_reviews[-1]
+
+    # Check if reviewer approved
+    if last_review.action == ActionType.APPROVED:
+        return (ReviewStatus.APPROVED, last_review.timestamp)
```
[the remark this answers is not in the document]
Comment (jpshackelford) [reply src/review/status.py:66]: Fixed in 2c8e545. Added check for commits after approval - status now correctly transitions to RE_REVIEW when author pushes new commits after approval.

File: tests/review/test_status.py [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+        ]
+
+        # Sort by status_priority
+        sorted_infos = sorted(infos, key=lambda x: x.status_priority)
+
+        # Expected order: REVIEW, RE_REVIEW, HOLD, APPROVED
+        assert sorted_infos[0].status == ReviewStatus.REVIEW
+        assert sorted_infos[1].status == ReviewStatus.RE_REVIEW
+        assert sorted_infos[2].status == ReviewStatus.HOLD
+        assert sorted_infos[3].status == ReviewStatus.APPROVED
```
[the remark this answers is not in the document]
Comment (jpshackelford) [reply tests/review/test_status.py:228]: Fixed in 2c8e545. Added `test_re_review_after_approval` test case that verifies the status transitions from APPROVED to RE_REVIEW when new commits are pushed after approval.

File: src/review/cli/list_cmd.py [hunk 11]
```diff
@@ -0,0 +1,226 @@
...
+                    f"({result.action_count} need action)[/]"
+                )
+            else:
+                console.print(f"\n[dim]Showing {len(result.reviews)} PRs needing your review[/]")
+
+            return 0
+
+    except Exception as e:
+        logger.debug("Full traceback:\n%s", traceback.format_exc())
+        console.print(f"[red]Error:[/] {e}")
```
[the remark this answers is not in the document]
Comment (jpshackelford) [reply src/review/cli/list_cmd.py:105]: Fixed in 2c8e545. Changed to `logger.exception()` which logs the full traceback at ERROR level when an exception is caught.

File: src/review/github_api.py [hunk 12]
```diff
@@ -0,0 +1,263 @@
...
+        Combines PRs from two queries:
+        1. PRs where reviewer is requested (pending reviews)
+        2. PRs reviewer has reviewed (for re-review/hold/approved detection)
+
+        Args:
+            reviewer: GitHub username (default: current user)
+            repos: List of "owner/repo" strings to filter by
+            author: Filter by PR author
+            limit: Maximum number of PRs to fetch
+            include_all: If True, include all PRs; if False, only actionable ones
```
[the remark this answers is not in the document]
Comment (jpshackelford) [reply src/review/github_api.py:75]: Acknowledged. The dual query approach with deduplication is acceptable for v1. Optimizing to a single GraphQL query can be explored in a future iteration if rate limits become a concern.

File: src/review/github_api.py [hunk 13]
```diff
@@ -0,0 +1,263 @@
...
+            repo=repo,
+            number=number,
+            title=title,
+            history=history,
+            status=status,
+            wait_seconds=wait_seconds,
+            ci_status=ci_status,
+            unresolved_thread_count=unresolved_thread_count,
+            author=author,
+            last_activity=last_activity,
```
[the remark this answers is not in the document]
Comment (jpshackelford) [reply src/review/github_api.py:328]: Valid refactoring suggestion. The function is cohesive (all steps needed to process a PR) but could benefit from extraction in a future cleanup. Acceptable for v1.

File: .pr/review-command-design.md [hunk 14]
```diff
@@ -0,0 +1,318 @@
...
+**Status colors:**
+- `review`, `re-review` → **yellow** (needs attention)
+- `hold`, `approved` → **dim** (info only, no action needed)
+
+**Wait time urgency:**
+- `> 48h` → **red** (critical)
+- `> 24h` → **yellow** (warning)  
+- `< 24h` → default
+
+**Sorting:**
```
[the remark this answers is not in the document]
Comment (jpshackelford) [reply .pr/review-command-design.md:155]: I have verified that the behavior of review command is correct as implemented. Sort with recent activity first. The pr command properly sorts by creation with newest on top. Eventually we may want option for these but that is for another day.

File: src/review/github_api.py [hunk 15]
```diff
@@ -0,0 +1,329 @@
...
+                    all_prs[key] = pr_data
+
+        # Process each PR to compute review status
+        reviews: list[ReviewInfo] = []
+        for pr_data in all_prs.values():
+            review_info = self._process_pr_for_reviewer(pr_data, reviewer, exclude_authors_lower)
+            if review_info:
+                reviews.append(review_info)
+
+        # Sort by most recently active first, then by status priority
```
[the remark this answers is not in the document]
Comment (jpshackelford) [reply src/review/github_api.py:133]: Design doc is incorrect. Implementation is correct.

**Comment by jpshackelford** [comment]
@openhands please study the PR as implemented and respond with squash merge conventional commit message
